test/system/500-networking.bats

```diff
@@ -116,4 +116,50 @@ load helpers
     fi
 }
 
+@test "podman network reload" {
+    skip_if_remote "podman network reload does not have remote support"
+    skip_if_rootless "podman network reload does not work rootless"
+
+    random_1=$(random_string 30)
+    HOST_PORT=12345
+    SERVER=http://127.0.0.1:$HOST_PORT
+
+    # Create a test file with random content
+    INDEX1=$PODMAN_TMPDIR/hello.txt
+    echo $random_1 > $INDEX1
+
+    # Bind-mount this file with a different name to a container running httpd
+    run_podman run -d --name myweb -p "$HOST_PORT:80" \
+               -v $INDEX1:/var/www/index.txt \
+               -w /var/www \
+               $IMAGE /bin/busybox-extras httpd -f -p 80
+    cid=$output
+
+    run_podman inspect $cid --format "{{.NetworkSettings.IPAddress}}"
+    ip="$output"
+    run_podman inspect $cid --format "{{.NetworkSettings.MacAddress}}"
+    mac="$output"
+
+    # Verify http contents: curl from localhost
+    run curl -s $SERVER/index.txt
+    is "$output" "$random_1" "curl 127.0.0.1:/index.txt"
+
+    ## FIXME: we should flush the iptables here
+    ## and check that curl fails so we can verify
+    ## that we actually do something in network reload
+
+    run_podman network reload $cid
+    is "$output" "$cid"
+
+    # check that we still have the same mac and ip
+    run_podman inspect $cid --format "{{.NetworkSettings.IPAddress}}"
+    is "$output" "$ip" "IP address changed after podman network reload"
+    run_podman inspect $cid --format "{{.NetworkSettings.MacAddress}}"
+    is "$output" "$mac" "MAC address changed after podman network reload"
+
+    # check that we can still curl
+    run curl -s $SERVER/index.txt
+    is "$output" "$random_1" "curl 127.0.0.1:/index.txt"
```

Comment:
Really nice test. May I suggest a final `run_podman rm -f $cid` for cleanup?

Reply:
Sure, I thought this would be done automatically

Reply:
It is, but with a warning message (which I am usually the only one to see). When I initially created this test framework I decided to use the default podman root (`/var/lib/containers` etc) instead of an ad-hoc `--root/--runroot` in a tmpdir; the reason was to closely reflect a default end-user setup. As part of that, I chose to require each test to clean up after itself, sort of as an extra level of confidence against unexpected side effects. It has actually been helpful at times in catching `rm` race conditions.